docs/source/distributed.fsdp.fully_shard.rst

```diff
@@ -0,0 +1,66 @@
+torch.distributed.fsdp.fully_shard
+==================================
+
+PyTorch FSDP2 (``fully_shard``)
+-------------------------------
+
+PyTorch FSDP2 provides a fully sharded data parallelism (FSDP) implementation
+targeting performant eager-mode while using per-parameter sharding for improved
+usability. Compared to PyTorch FSDP1 (``FullyShardedDataParallel``):
+
+- FSDP2 uses ``DTensor``-based dim-0 per-parameter sharding for a simpler
+  sharding representation compared to FSDP1's flat-parameter sharding, while
+  preserving similar throughput performance. This provides a more intuitive
+  user experience, relaxes constraints around frozen parameters, and allows for
+  communication-free sharded state dicts.
```

Comment:
nit: would like to say more about the drawbacks of flat-param sharding, esp. RE state-dicts.  again, maybe OK to link to another wiki or post where this is more in depth.  (but in particular, this is a motivating factor for switching)